query/src/test/scala/filodb/query/exec/RemoteMetadataExecSpec.scala

```diff
@@ -31,27 +39,94 @@ class RemoteMetadataExecSpec extends AnyFunSpec with Matchers with ScalaFutures
+                ishard: Int): Unit = {
+    val partTagsUTF8s = partKeyLabelValues.map{case (m, t) => (m,  t.map { case (k, v) => (k.utf8, v.utf8)})}
+    // NOTE: due to max-chunk-size in storeConf = 100, this will make (numRawSamples / 100) chunks
+    // Be sure to reset the builder; it is in an Object so static and shared amongst tests
+    builder.reset()
+    partTagsUTF8s.map { case (metric, partTagsUTF8) =>
+      tuples.map { t => SeqRowReader(Seq(t._1, t._2, metric, partTagsUTF8)) }
+        .foreach(builder.addFromReader(_, Schemas.promCounter))
+    }
+    memStore.setup(timeseriesDatasetMultipleShardKeys.ref, Schemas(Schemas.promCounter), ishard, TestData.storeConf)
+    memStore.ingest(timeseriesDatasetMultipleShardKeys.ref, ishard, SomeData(builder.allContainers.head, 0))
+  }
+
   val jobQueryResult1 = ArrayBuffer(("job", "myCoolService"), ("unicode_tag", "uni\u03BCtag"))
   val jobQueryResult2 = Array("http_req_total", "http_resp_time")
   val jobQueryResult3 = Array("job", "__name__", "unicode_tag", "instance")
 
+  override def beforeAll(): Unit = {
+    for (ishard <- 0 until shardPartKeyLabelValues.size) {
+      initShard(memStore, shardPartKeyLabelValues(ishard), ishard)
+    }
+    memStore.refreshIndexForTesting(timeseriesDatasetMultipleShardKeys.ref)
+  }
+
+  override def afterAll(): Unit = {
+    memStore.shutdown()
+  }
+
+  val executeDispatcher = new PlanDispatcher {
+    override def isLocalCall: Boolean = ???
+    override def clusterName: String = ???
+    override def dispatch(plan: ExecPlan)
+                         (implicit sched: Scheduler): Task[QueryResponse] = {
+      plan.execute(memStore, querySession)(sched)
+    }
+  }
+
   implicit val testingBackend: SttpBackend[Future, Nothing] = SttpBackendStub.asynchronousFuture
     .whenRequestMatches(request =>
       request.uri.path.startsWith(List("api","v1","label")) && request.uri.path.last == "values"
     )
     .thenRespondWrapped(Future {
-      Response(Right(Right(MetadataSuccessResponse(Seq(LabelSampl(Seq("http_req_total", "http_resp_time"))), "success", Option.empty, Option.empty))), StatusCodes.PartialContent, "", Nil, Nil)
+      Response(Right(Right(MetadataSuccessResponse(Seq(LabelSampl("http_req_total"), LabelSampl("http_resp_time")), "success", Option.empty, Option.empty))), StatusCodes.PartialContent, "", Nil, Nil)
     })
     .whenRequestMatches(request =>
       request.uri.path.startsWith(List("api","v1","series"))
     )
     .thenRespondWrapped(Future {
-      Response(Right(Right(MetadataSuccessResponse(Seq(MetadataSampl(Map(("job" -> "myCoolService"), ("unicode_tag" -> "uniμtag")))), "success", Option.empty, Option.empty))), StatusCodes.PartialContent, "", Nil, Nil)
+      Response(Right(Right(MetadataSuccessResponse(Seq(MetadataMapSampl(Map(("job" -> "myCoolService"), ("unicode_tag" -> "uniμtag")))), "success", Option.empty, Option.empty))), StatusCodes.PartialContent, "", Nil, Nil)
```

Comment:
Can you also add a test case with empty array for `LabelSampl` and `MetadataMapSampl`.

Reply:
added testcases for empty results